code/modules/vehicles/vehicle.dm

```diff
@@ -0,0 +1,403 @@
+				lastmove = 0
+
+/obj/vehicle/MouseDrop_T(mob/M, mob/user)
+	if (!user.canUseTopic(src) || (user != M))
+		return
+	if(!ishuman(user)) // no silicons or drones in vehicles.
+		return
+	if(user.buckled)
+		user << "<span class='warning'>You are currently buckled and cannot move.</span>"
+		return
+	if(occupants == occupants_max && driver)
+		user << "<span class='warning'>The [src.name] is full!</span>"
+		return
+	for(var/mob/living/simple_animal/slime/S in range(1,user))
+		if(S.Victim == user)
+			user << "<span class='warning'>You're too busy getting your life sucked out of you!</span>"
+			return
+
+	visible_message("[user] starts to climb into [src.name].")
+
+	if(do_after(user, 40, target = src))
+		if(user.buckled)
+			user << "<span class='warning'>You have been buckled and cannot move.</span>"
+		else
+			moved_inside(user)
+	else
+		user << "<span class='warning'>You stop entering the vehicle!</span>"
+	return
+
+/obj/vehicle/MouseDrop(atom/over)
+	if(over != usr)
+		return
+	interact(usr)
+
+/obj/vehicle/MouseDrop(over_object, src_location, over_location)
+	..()
+	if(over_object == usr && Adjacent(usr) && (in_range(src, usr) || usr.contents.Find(src)))
+		interact(usr)
+
+/obj/vehicle/interact(mob/user)
+	if(user.incapacitated() || user.lying || !Adjacent(user))
+		return
+	user.face_atom(src)
+	var/dat = "<h3>Vehicle menu</h3>"
+	for(var/obj/item/P in parts)
+		dat += build_text(P)
+	dat += "</div>"
+
+	var/datum/browser/popup = new(user, "vehicle", src.name, 500, 500)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/vehicle/Topic(href, href_list)
+	if(usr.stat || !Adjacent(usr) || usr.lying)
+		return
+	if(href_list["parts"])
+		var/obj/item/O = locate(href_list["parts"])
+		var/obj/item/vehicle_parts/P = O
+		if(locked)
+			usr << "<span class ='warning'>the [src] is locked!</span>"
+			return
+		if(driver || occupants)
+			usr << "<span class = 'warning'>You can't modify it while there are people inside!</span>"
+			return
+		if(P.part_type == "chassis")
+			usr << "<span class ='notice'>You begin to completely dismantle [src]</span>"
+			if(do_after(usr, 500, target = src))
+				for(var/obj/item/L in parts)
+					L.loc = get_turf(usr)
+				usr << "<span class ='notice'>You completely dismantle [src]</span>"
+				qdel(src)
+			else
+				usr << "<span class ='notice'>You stop dismantling [src]</span>"
+		else
+			usr << "<span class ='notice'>You begin removing [P]</span>"
+			if(do_after(usr, 40, target = src))
+				if(P.part_type == "seat")
+					occupants_max--
+				P.loc = get_turf(usr)
+				parts -= P
+				if(P.icon_state in overlays)
+					overlays -= P.icon_state
+				update_stats()
+				usr << "<span class ='notice'>You sucessfully removed [P]</span>"
+			else
+				usr << "<span class ='warning'>You stop removing [P]</span>"
+			interact(usr)
+
+/obj/vehicle/proc/build_text(obj/item/P)
+	. = ""
+	var/name_text = ""
+
+	name_text ="<A href='?src=\ref[src];parts=\ref[P]'>[P.name]</A>"
+	. = "[name_text]<BR>"
```

Comment:
Whilst not functionally and theoretically not syntactically incorrect, should be lowercase to remain consistent.